src/test/java/org/jenkinsci/plugins/workflow/flow/FlowExecutionListTest.java

```diff
@@ -160,6 +165,58 @@ public class FlowExecutionListTest {
         });
     }
 
+    @Ignore("Build never completes due to infinite loop")
+    @LocalData
+    @Test public void stepExecutionIteratorDoesNotLeakBuildsWhenOneIsStuckInfiniteLoop() throws Throwable {
```

Comment:
I am not sure yet if I want to do anything about infinite loops directly. If so, I will split this into a new PR and test various forms of infinite loops, and if not, I will just delete this test since the other one is simpler and tests the memory leak.

Reply:
Moved into #349 and removed here in [7100611](https://github.com/jenkinsci/workflow-api-plugin/pull/347/commits/71006111e7df33fbcc0cde1e8a81569127953492).